### PR TITLE
Authorization branch PR

### PR DESCRIPTION
**Pull request by mertkantaroglu**
Hello,

In this branch me and @rica213;

- Installed CanCanCan in the project.
- Added a `role` column to the users table by using a migration.
- Added functionality where the user can delete a post if it is theirs or if they have an admin role. Used CanCanCan for this authorization.
    - Implemented the post deleting functionality. Added the "Delete" button to the view and made sure that only authorized users can see it.
- Added functionality where the user can delete a comment if it is theirs or if they have an admin role. Used CanCanCan for this authorization.
    - Implemented the comment deleting functionality. Added the "Delete" button to the view and made sure that only authorized users can see it.
- Fixed linter errors

Looking forward to hearing from you.

Thanks.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -14,4 +14,8 @@ class User < ApplicationRecord
   def last_posts
     posts.order(created_at: :desc).limit(3)
   end
+
+  def admin?
+    role == 'admin'
+  end
```
Comment (CollinsTatang) [inline app/models/user.rb:20]: Great job

- Creating your admin role method in the User model :+1:

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -11,16 +11,26 @@ def new
...
+
+  def destroy
     @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:id])
+
+    @post.likes.destroy_all
+    @post.comments.destroy_all
+    @post.destroy
+
+    redirect_to user_posts_path(@user)
```
Comment (CollinsTatang) [inline app/controllers/posts_controller.rb:35]: - [x] Kindly make use of **CanCanCan** for Post authorization in the Post Controller before any action.
```ruby
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @mertkantaroglu , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (mertkantaroglu) [reply app/models/user.rb:20]: Thank you @CollinsTatang

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Your project meets all changes requested by the previous reviewer ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**